pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/FunctionCommon.java

```diff
@@ -279,18 +279,22 @@ public static File createPkgTempFile() throws IOException {
     }
 
     public static File extractFileFromPkgURL(String destPkgUrl) throws IOException, URISyntaxException {
+        final File pkgTempFile = createPkgTempFile();
+        pkgTempFile.deleteOnExit();
```

Comment:
this is not guaranteed to happen.
but it is already like this in legacy code, we can keep it